pkg/hive/hive.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hive
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/hive/pb"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	protocolName    = "hive"
+	protocolVersion = "1.0.0"
+	peersStreamName = "peers"
+	maxPO           = 7
+	messageTimeout  = 5 * time.Second // maximum allowed time for a message to be read or written.
+)
+
+type Service struct {
+	streamer          p2p.Streamer
+	connectionManager ConnectionManager
+	peerSuggester     PeerSuggester
+	addressFinder     AddressFinder
+	logger            logging.Logger
+}
+
+type Options struct {
+	Streamer          p2p.Streamer
+	ConnectionManager ConnectionManager
+	PeerSuggester     PeerSuggester
+	AddressFinder     AddressFinder
+	Logger            logging.Logger
+}
+
+func New(o Options) *Service {
+	return &Service{
+		streamer:          o.Streamer,
+		logger:            o.Logger,
+		connectionManager: o.ConnectionManager,
+		peerSuggester:     o.PeerSuggester,
+		addressFinder:     o.AddressFinder,
+	}
+}
+
+type BzzAddress struct {
```

Comment:
I think that BzzAddress can be unexported.

Reply:
What do you guys think about removing the BzzAddress and using just string underlay for now?

Reply:
I agree to remove it. I also do not see that we need it.